Firebase/android/app/src/main/java/com/novoda/bonfire/channel/view/ChannelsAdapter.java

```diff
@@ -39,18 +39,18 @@ public long getItemId(int position) {
         return channels.getChannelAt(position).hashCode();
     }
 
-    public void attach(ChannelsDisplayer.ChannelSelectionListener channelSelectionListener) {
-        this.channelSelectionListener = channelSelectionListener;
+    public void attach(ChannelsDisplayer.ChannelsInteractionListener channelsInteractionListener) {
+        this.channelsInteractionListener = channelsInteractionListener;
     }
 
-    public void detach(ChannelsDisplayer.ChannelSelectionListener channelSelectionListener) {
-        this.channelSelectionListener = ChannelsDisplayer.ChannelSelectionListener.NO_OP;
+    public void detach(ChannelsDisplayer.ChannelsInteractionListener channelsInteractionListener) {
+        this.channelsInteractionListener = ChannelsDisplayer.ChannelsInteractionListener.NO_OP;
```

Comment:
no_op can be removed and the listener can be renamed back to ChannelSelectionListener